### PR TITLE
feat: create angular publish pipeline

### PR DESCRIPTION
**Pull request by stuart-rimel**
### :page_facing_up: Summary of Changes

Adds angular integration to our publish pipeline. Also adds a PR check for the angular integration build errors.

Changes:
- Creates `publish-angular.yml` pipeline
- Creates `build-angular.yml` and includes as PR check in `merge-gate.yml`
- Fixes some bugs in `publish-react.yml`

### :thought_balloon: Type of Change

- [x] Bug fix (fixes an issue)
- [x] New feature (adds functionality)
- [ ] Configuration change (modifies scaling or properties of existing infrastructure & services)
- [ ] Documentation change
- [ ] Other

### :clipboard: Test Plan

Tested with manual pipeline runs

### :white_check_mark: Self Code Review Checklist

PR authors and reviewers, please verify that all of these items have been completed.

- [x] My code is clean and readable
- [x] I followed standard conventions
- [ ] Component code is WCAG 2.2 compliant (if applicable)
- [ ] I have made theme files changes (if applicable)
- [ ] I have made documentation changes (if applicable)
- [ ] I have made Storybook changes including usage documentation (if applicable)
- [ ] I have tested the component thoroughly in Storybook including RTL rendering (if applicable)

### :link: Azure DevOps Work Item

[AB#570772](https://dev.azure.com/ViewpointVSO/74194628-0b6e-4b03-95a4-2d06c069dcbe/_workitems/edit/570772)

### CODE REVIEW

File: .github/workflows/build-angular.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ vars.NODE_VERSION }}
```
Comment (Kai-Richardson) [inline .github/workflows/build-angular.yml:20]: should set npm cache here

File: .github/workflows/build-angular.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Build Stencil components
+        id: stencil
+        run: npm run build:ci
+        continue-on-error: false
```
Comment (Kai-Richardson) [inline .github/workflows/build-angular.yml:28]: this is unnecessary for all these steps, this is the default

File: .github/workflows/build-angular.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ vars.NODE_VERSION }}
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Build Stencil components
+        id: stencil
```
Comment (Kai-Richardson) [inline .github/workflows/build-angular.yml:26]: i don't see why we're setting IDs if we're not going to be using them

**Review by gabriel-piltzer-trimble — APPROVED** [review]

File: .github/workflows/build-angular.yml (hunk 1 above)
Comment (stuart-rimel) [reply .github/workflows/build-angular.yml:20]: Set in [new PR](https://github.com/Trimble-Construction/modus-wc-2.0/pull/58)

File: .github/workflows/build-angular.yml (hunk 2 above)
Comment (stuart-rimel) [reply .github/workflows/build-angular.yml:28]: Removed in [new PR](https://github.com/Trimble-Construction/modus-wc-2.0/pull/58)

File: .github/workflows/build-angular.yml (hunk 3 above)
Comment (stuart-rimel) [reply .github/workflows/build-angular.yml:26]: Removed in [new PR](https://github.com/Trimble-Construction/modus-wc-2.0/pull/58)

**Pull request merged**